src/binary/paths.ts

```diff
@@ -1,35 +1,56 @@
 import * as path from "path";
-import {
-  ACTIVE_PATH,
-  BINARY_ROOT_PATH,
-  BINARY_UPDATE_URL,
-  BINARY_UPDATE_VERSION_FILE_URL,
-} from "../consts";
+import * as vscode from "vscode";
+import * as fs from "fs";
+import { BINARY_UPDATE_URL, BINARY_UPDATE_VERSION_FILE_URL } from "../consts";
 
+let binaryRootPath: string | undefined;
 const ARCHITECTURE = getArch();
 const SUFFIX = getSuffix();
 const BUNDLE_SUFFIX = getBundleSuffix();
 
+export function setBinaryRootPath(updatedPath: vscode.Uri): void {
+  binaryRootPath = path.join(updatedPath.fsPath, "binaries");
+  // eslint-disable-next-line no-debugger
+  debugger;
+  fs.mkdir(binaryRootPath, { recursive: true }, () => {});
```

Comment:
mkdir is an async callback api, use `await fs.promises.mkdir`, if you want to be sure it is created